src/rubicon/objc/api.py

```diff
@@ -91,6 +91,21 @@
 # the Python objects are not destroyed if they are otherwise no Python references left.
 _keep_alive_objects = {}
 
+# Methods that return an object which is implicitly retained by the caller.
+# See https://clang.llvm.org/docs/AutomaticReferenceCounting.html#semantics-of-method-families.
+_RETURNS_RETAINED_FAMILIES = {"init", "alloc", "new", "copy", "mutableCopy"}
+
+
+def _get_method_family(method_name: str) -> str:
+    # See https://clang.llvm.org/docs/AutomaticReferenceCounting.html#method-families.
+    method_name = method_name.lstrip("_").split(":")[0]
+    leading_lowercases = []
+    for c in method_name:
+        if c.isupper():
+            break
+        leading_lowercases.append(c)
+    return "".join(leading_lowercases)
```

Comment:
The implementation here looks fine; since we can't guarantee we're actually exercising it fully, we should probably add some explicit unit tests that we're catching all the possible naming strategies that it should match.

Reply:
Good point.